docs/source/tracking/data-api.rst

```diff
@@ -0,0 +1,227 @@
+
+    dataset_source_url = "https://raw.githubusercontent.com/mlflow/mlflow/master/tests/datasets/winequality-white.csv"
+    raw_data = pd.read_csv(dataset_source_url, delimiter=CSV_DELIMITER)
+
+    # Create an instance of a PandasDataset
+    dataset = mlflow.data.from_pandas(
+        raw_data, source=dataset_source_url, name="wine quality - white", targets="quality"
+    )
+
+    # Log the Dataset to an MLflow run by using the `log_input` API
+    with mlflow.start_run() as run:
+        mlflow.log_input(dataset, context="training")
+
+    # Retrieve the run information
+    logged_run = mlflow.get_run(run.info.run_id)
+
+    # Retrieve the Dataset object
+    logged_dataset = logged_run.inputs.dataset_inputs[0].dataset
+
+    # View some of the recorded Dataset information
+    print(f"Dataset name: {logged_dataset.name}")
+    print(f"Dataset digest: {logged_dataset.digest}")
+    print(f"Dataset profile: {logged_dataset.profile}")
+    print(f"Dataset schema: {logged_dataset.schema}")
+
+
+The stdout results of the above code snippet are as follows:
+
+.. code-block:: shell
+
+    Dataset name: wine quality - white
+    Dataset digest: 2a1e42c4
+    Dataset profile: {"num_rows": 4898, "num_elements": 58776}
+    Dataset schema: {"mlflow_colspec": [
+        {"type": "double", "name": "fixed acidity"}, 
+        {"type": "double", "name": "volatile acidity"}, 
+        {"type": "double", "name": "citric acid"}, 
+        {"type": "double", "name": "residual sugar"}, 
+        {"type": "double", "name": "chlorides"}, 
+        {"type": "double", "name": "free sulfur dioxide"}, 
+        {"type": "double", "name": "total sulfur dioxide"}, 
+        {"type": "double", "name": "density"}, 
+        {"type": "double", "name": "pH"}, 
+        {"type": "double", "name": "sulphates"}, 
+        {"type": "double", "name": "alcohol"}, 
+        {"type": "long", "name": "quality"}
+        ]}
+
+We can navigate to the MLflow UI to see what this looks like for a logged Dataset as well. 
+
+.. figure:: ../_static/images/tracking/dataset-mlflow-ui.png
+    :align: center 
+    :figwidth: 100%
+
+When we want to load the dataset back from the location that it's stored (calling ``load`` will download the data locally), we 
+access the Dataset's source via the following API:
+
+.. code-block:: python
+
+   # Loading the dataset's source
+   dataset_source = mlflow.data.get_source(logged_dataset)
+
+   local_dataset = retrieved_data.load()
+
+   print(f"The local file where the data has been downloaded to: {local_dataset}")
+
+   # Load the data again
+   loaded_data = pd.read_csv(local_dataset, delimiter=CSV_DELIMITER)
+
+The print statement from above resolves to the local file that was created when calling ``load``.
+
+.. code-block:: shell
+
+    The local file where the data has been downloaded to:
+    /var/folders/cd/n8n0rm2x53l_s0xv_j_xklb00000gp/T/tmpuxwtrul1/winequality-white.csv
+
+What is the purpose of ``mlflow.data``?
```

Comment:
What about rephrasing this to sth like `Using the Dataset with Other MLflow Features`? This section sound like explaining the concrete usage rather than purpose.